### PR TITLE
Dev - now with dynamic progress bar !

### PR DESCRIPTION
**Pull request by sofiiagran**
## Description
This is a small PR where the progress bar now is dynamic

## Changes

Added a function to calculate how much of a course the user have completed, called "checkProgressCourse" and "checkProgressSection"

## Checklist

- [x] Code has been tested locally and passes all relevant tests.
- [x] Documentation has been updated to reflect the changes, if applicable.
- [x] Code follows the established coding style and guidelines of the project.
- [x] All new and existing tests related to the changes have passed.
- [x] Any necessary dependencies or new packages have been properly documented.
- [x] Pull request title and description are clear and descriptive.
- [x] Reviewers have been assigned to the pull request.
- [x] Any potential security implications have been considered and addressed.
- [x] Performance impact of the changes has been evaluated, if relevant.

## If mobile/frontend pull request, what version of the backend is it stable, and running on? 

Staging backend
Because we still lack database refactoring, it can only be tested on the last section of the first course.

## Notes for Reviewers

More tests to come later to day I promise 💯 👍

### CODE REVIEW

**Comment by sofiiagran** [comment]
@PrometheusBork now fixed

**Comment by sofiiagran** [comment]
@Donhusum can you please look at the implementation of getSection in StorageServices? does it look good/right?

**Comment by sofiiagran** [comment]
Snapshots fail... but I cannot update it on my computer...

**Review by PrometheusBork — APPROVED** [review]
Very nice good work! Tested on my device it works fine!

**Review by Donhusum — CHANGES_REQUESTED** [review]
I have written some comment on changes. I am also trying to test your branch on my device. I have some problems there, but i am usurer if it is due to the changes, or the fact that the DB have been semi purged.

File: eml/test/screens/section/__snapshots__/SectionScreen.test.jsx.snap [hunk 1]
```diff
@@ -590,6 +593,9 @@ exports[`SectionScreen Renders section screen with sections loaded 1`] = `
                     "paddingBottom": 15,
                     "paddingTop": 15,
                   },
+                  {
+                    "backgroundColor": "#8DD08C",
+                  },
```
[the remark this answers is not in the document]
Comment (sofiiagran) [reply eml/test/screens/section/__snapshots__/SectionScreen.test.jsx.snap:598]: jeg tror det er lige meget i tests?

[the remark this answers is not in the document]
Comment (PrometheusBork) [reply eml/test/screens/section/__snapshots__/SectionScreen.test.jsx.snap:598]: Snapshot so it doesn't matter

File: eml/services/StorageService.js [hunk 2]
```diff
@@ -171,6 +178,70 @@ const refreshCourseList = async (courseList) => {
...
+		} catch (e){
+			if (e?.response?.data != null) {
+				throw new Error('Error in getSection: ', e.response.data);
+			} else {
+				throw new Error('Error in getSection: ', e);
+			}
+		}
+	} finally {
+		return await refreshSection(section);
+	}
```
[the remark this answers is not in the document]
Comment (sofiiagran) [reply eml/services/StorageService.js:209]: Have I not modified it enough?

File: eml/test/screens/section/__snapshots__/SectionScreen.test.jsx.snap (hunk 1 above)
[the remark this answers is not in the document]
Comment (Donhusum) [reply eml/test/screens/section/__snapshots__/SectionScreen.test.jsx.snap:598]: Sorry, I did not see it was a snapshot. Ignore this comment

**Comment by Donhusum** [comment]
> I have written some comment on changes. I am also trying to test your branch on my device. I have some problems there, but i am usurer if it is due to the changes, or the fact that the DB have been semi purged.

It seem to have been problems do to the DB purge. I all works fine now.

File: eml/services/StorageService.js (hunk 2 above)
[the remark this answers is not in the document]
Comment (Donhusum) [reply eml/services/StorageService.js:209]: Actually I think you have, my bad.

File: eml/services/utilityFunctions.js [hunk 3]
```diff
@@ -204,13 +206,58 @@ function isFirstAttemptExercise(student, compId) {
...
+	progress = (progress / totalComponents) * 100;
+
+	return progress;
+}
+
+// Returns the students progress of a section
+export async function checkProgressSection(sectionId) {
+	const userId = await StorageService.getUserId();
+	const student = await userApi.getStudentInfo(userId);
+	const section = await StorageService.getSection(sectionId);
```
[the remark this answers is not in the document]
Comment (sofiiagran) [reply eml/services/utilityFunctions.js:236]: will be fixed on new commit.

**Comment by sofiiagran** [comment]
New commit in 20 min, waiting for DB to be restored, right now it is empty and cannot be tested

**Review by Donhusum — APPROVED** [review]

**Pull request merged**